Source/WebCore/loader/cache/CachedCSSStyleSheet.cpp

```diff
@@ -75,7 +75,9 @@ String CachedCSSStyleSheet::encoding() const
 
 const String CachedCSSStyleSheet::sheetText(MIMETypeCheckHint mimeTypeCheckHint, bool* hasValidMIMEType) const
 {
-    if (!m_data || m_data->isEmpty() || !canUseSheet(mimeTypeCheckHint, hasValidMIMEType))
+    // Check MIME type first, to ensure that if it's not text/css, we don't load the resource.
+    // https://html.spec.whatwg.org/#link-type-stylesheet:process-the-linked-resource
+    if (!canUseSheet(mimeTypeCheckHint, hasValidMIMEType) || !m_data || m_data->isEmpty())
```

Comment:
I don't understand this change at all. How does checking canUseSheet() first prevent loading the resource? Checking `m_data` or `m_data->isEmpty()` doesn't trigger a load.

Reply:
Is it to guarantee that `hasValidMIMEType` gets set even if `!m_data || m_data->isEmpty()`? If so, please clarify in the associated code comment.

Reply:
Thanks yeah I can see now the comment was misleading. Hopefully the updated one is a more accurate explanation